### PR TITLE
specify command and extra files in project.clj

### PR DESCRIPTION
**Pull request by mccraigmccraig**
allows config to be given in project.clj with the form :

:uberimage {:cmd ["/bin/dash" "/myrunscript" "param1" "param2"]
                   :files {"myrunscript" "docker/myrunscript"}}

the :cmd maps directly to the Dockerfile CMD statement, and :files is a map of
additional files to be copied into the docker image, with docker image paths as keys
and lein project source paths as values

this permits more flexibility in starting the containerized program, to e.g.
configure extra environment with a script

an example : i have a need to discover the container's gateway address, because there is an HAProxy listening there, fronting one of the containerized programs service dependencies : it's trivial to do if i can add a script to the docker image, and exec the uberjar from that script

### CODE REVIEW

**Comment by hugoduncan** [comment]
Looks like great functionality!

Sorry to go on about commit messages, but please use full sentences and puctuation, and wrap at 72 characters.  I added some links in the new [contributing guidelines](CONTRIBUTING.md).

File: src/leiningen/uberimage.clj [hunk 1]
```diff
@@ -15,22 +15,31 @@
...
+                "ADD uberjar.jar uberjar.jar"
+                (str "CMD [" cmd-str "]")
+                (for [[tar-path local-path] files] (str "ADD " tar-path " " tar-path)))
+         (string/join "\n"))))
 
 (defn buildtar
   "Return an InputStream that will deliver a tar archive with a Dockerfile
   and the uberjar."
-  [tar-output-stream piped-output-stream standalone-filename options]
+  [tar-output-stream piped-output-stream standalone-filename {:keys [files] :as options}]
```
Comment (hugoduncan) [inline src/leiningen/uberimage.clj:33]: Please wrap at less than 80 chars

File: src/leiningen/uberimage.clj [hunk 2]
```diff
@@ -15,22 +15,31 @@
...
-  (format "FROM %s
-ADD uberjar.jar uberjar.jar
-CMD [\"/usr/bin/java\", \"-jar\", \"/uberjar.jar\"]"
-          base-image))
+  [{:keys [cmd files base-image]}]
+  (let [cmd (or cmd ["/usr/bin/java" "-jar" "/uberjar.jar"])
+        cmd (if (sequential? cmd) cmd [cmd])
+        cmd-str (->> (for [s cmd] (str "\"" s "\""))
+                     (string/join ","))]
+    (->> (apply vector
```
Comment (hugoduncan) [inline src/leiningen/uberimage.clj:23]: maybe `(concat […] …)` would be clearer.

**Comment by mccraigmccraig** [comment]
ok, i've closed this PR and opened a new one on a topic-branch

**Pull request closed without merging**